compiler/src/test/java/com/squareup/anvil/compiler/dagger/BindsMethodValidatorTest.kt

```diff
@@ -352,17 +353,43 @@ class BindsMethodValidatorTest(
     }
   }
 
+  @Test
+  fun `binding inside interface is valid`() {
+    compile(
+      """
+      package com.squareup.test
+
+      import dagger.Binds
+      import dagger.Module
+      import javax.inject.Inject
+
+      interface Foo : Bar
+      interface Bar
+
+      @Module
+      interface BarModule {
+        @Binds
+        fun bindsBar(foo: Foo): Bar
+      }
+      """,
+    ) {
+      assertThat(exitCode).isEqualTo(OK)
+    }
+  }
+
   private fun compile(
     @Language("kotlin") vararg sources: String,
     previousCompilationResult: JvmCompilationResult? = null,
     enableDagger: Boolean = useDagger,
+    forceEmbeddedMode: Boolean = false,
```

Comment:
Can you add a small comment explaining why we need this for posterity?

Reply:
I'm not sure how to explain it in a comment.
I've added this flag to fix tests according to your suggestion: https://github.com/square/anvil/pull/831#issuecomment-1869815259